MDANSE_GUI/Src/MDANSE_GUI/MolecularViewer/MolecularViewer.py

```diff
@@ -488,38 +565,40 @@ def get_render_window(self):
     def iren(self):
         return self._iren
 
-    def on_change_atomic_trace_opacity(self, opacity):
+    def on_change_atomic_trace_opacity(self, surface_number, opacity):
         """Event handler called when the opacity level is changed."""
 
-        if self._surface is None:
+        if len(self._surfaces) <= surface_number:
```

Comment:
Because we want to check it's out of range it makes more sense to me this way round. However `len(self._surfaces)` is 1 more than the largest index, so are you sure you wanted `<=`?
```suggestion
        if surface_number > len(self._surfaces):
```

Reply:
I swapped the variables around.

An example: an array of length 3 has valid indices 0, 1, 2. Indices 3, 4, ... are out of range. This will be captured by the condition `index >= length`.